src/compiler/target/javascript/mod.rs

```diff
@@ -68,9 +68,50 @@ impl JavaScript {
       ActionType::Variable(res) => self.action_var(res, lb),
       ActionType::VarRef(res) => lb.code(res + if inline { "" } else { ";" }),
       ActionType::While(res) => self.action_while(res, lb),
-      ActionType::If(_, _, _) => unimplemented!(), // TODO: make this
+      ActionType::If(if_, else_if, else_) => self.action_if(if_, else_if, else_, lb), // TODO: make this
     };
   }
+  pub fn action_if(
+    &mut self, 
+    if_: (std::boxed::Box<tokenize::action::Action>, tokenize::actions::Actions), 
+    else_if: std::vec::Vec<(tokenize::action::Action, tokenize::actions::Actions)>,
+    else_: std::option::Option<tokenize::actions::Actions>,
+    lb: &mut impl BuildItems
+  ) {
+    // if
+    let statement = *if_.0;
+    let mut prefix = Inline::from_str("if (");
+    self.action(statement, &mut prefix, true);
+    prefix.code(")");
+    let mut actions = Block::new();
+    for action in if_.1.list {
+      self.action(action, &mut actions, false);
+    }
+    lb.function(prefix, actions);
```

Comment:
We repeat this code multiple times here, might be worth it to move it to a function to reduce the amount of code.

Something like this:  
```rust
fn if_block(lb: &mut impl BuildItems, body: Actions, start_prefix_with: &'static str, add_to_prefix: impl FnOnce(prefix: Inline)) {
    let statement = *if_.0;
    let mut prefix = Inline::from_str(start_prefix_with);
    add_to_prefix(prefix);

    let mut actions = Block::new();
    for action in body {
      self.action(action, &mut actions, false);
    }
    lb.function(prefix, actions);
}
```

The function can be called using:
```rust
let statement = *if_.0;
if_block(lb, if_.1.list, "if (", move |prefix| {
    self.action(statement, &mut prefix, true);
    prefix.code(")");
})
```

Reply:
This is appearing as invalid (line `1`). I've never seen inline implementations and not sure how they work. 

```rust
impl FnOnce(prefix: Inline)
```

Reply:
These inline functions can be a bit tricky, if it doesn't work you might want to try some of these:  
- not use the move keyword
- add the `Inline` type to the inline function
- the body argument to a reference
- change the body argument to a implementation of iterator, and add .iter() to the if_1.list

Reply:
What do you think of this? 07178ae28aad2c2acf3107f274071d7f9bf6f038

Reply:
I'll checkout this branch and see if i can implement it :)

Reply:
Looks much better. Thanks :-)